src/gui/editorwidgets/qgsvaluerelationwidgetwrapper.cpp

```diff
@@ -157,11 +157,15 @@ void QgsFilteredTableWidget::populate( QgsValueRelationFieldFormatter::ValueRela
 
 void QgsFilteredTableWidget::setIndeterminateState()
 {
-  for ( int j = 0; j < mTableWidget->rowCount(); j++ )
+  for ( int rowIndex = 0; rowIndex < mTableWidget->rowCount(); rowIndex++ )
   {
-    for ( int i = 0; i < mColumnCount; ++i )
+    for ( int columnIndex = 0; columnIndex < mColumnCount; ++columnIndex )
     {
-      whileBlocking( mTableWidget )->item( j, i )->setCheckState( Qt::PartiallyChecked );
+      if ( rowIndex * mColumnCount + columnIndex >= mCache.count() )
```

Comment:
Is the problem that "item" returns null here? It might be cleaner to just go:

   if ( currentItem = item( rowIndex, columnIndex) ) { ... }

Reply:
Yes, that's the problem, but I find it cleaner this way: why would we want to iterate over a non-existing table item when we already know in advance that the index is overflowing?

Reply:
Well the logic for sizing mCache and building checkbox widgets based on the size of mCache is somewhere else entirely, and there's a good chance that if someone messes with that logic they won't realise that this code even exists and needs updating accordingly.